Identity_Infrastructure/Repositories/GenericRepository.cs

```diff
@@ -0,0 +1,125 @@
+﻿using Identity_Application.Interfaces.Repository;
+using Identity_Domain.Entities.Base;
+using Microsoft.EntityFrameworkCore;
+using System.Linq.Expressions;
+
+namespace Identity_Infrastructure.Repositories;
+
+public class GenericRepository<T> : IGenericRepository<T> where T : Entity
```

Comment:
I'm not sure if we need it in each method, I mean try catch
if you want we can discuss it in call

Reply:
Yes